tests/unit/test_client.py

```diff
@@ -1132,11 +1225,19 @@ def test_create_routine_w_conflict(self):
         conn = client._connection = make_connection(
             google.api_core.exceptions.AlreadyExists("routine already exists")
         )
+        path = "/projects/test-routine-project/datasets/test_routines/routines"
         full_routine_id = "test-routine-project.test_routines.minimal_routine"
         routine = Routine(full_routine_id)
 
         with pytest.raises(google.api_core.exceptions.AlreadyExists):
-            client.create_routine(routine)
+            with mock.patch(
+                "google.cloud.bigquery.opentelemetry_tracing._get_final_span_attributes"
+            ) as final_attributes:
+                client.create_routine(routine)
+
+            final_attributes.assert_called_once_with(  # pragma: NO COVER
```

Comment:
I think you should be able to cover this line if outside of the `pytest.raises` block.

```suggestion
        final_attributes.assert_called_once_with(
```

Reply:
Also, this might be a good test to verify that the status code is set as expected.